src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.cs

```diff
@@ -563,15 +593,15 @@ private static string GetResourceString(ExceptionResource resource, int currentD
             return message;
         }
 
-        public static FormatException GetFormatException()
+        [DoesNotReturn]
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
Marking these methods with `NoInlining` is de-optimization. It is better to let the JIT to try inlining the method (that is going to fail), but it allows the JIT to observe that the method throws exception and never returns.

Reply:
Ok, I'll remove this from all stub ThrowHelpers since we're a bit inconsistent currently. 

> Marking these methods with NoInlining is de-optimization.

Can we make that assumption for older versions of the JIT as well? Asking since we're also targeting `netstandard2.0` here. Granted our optimizations are generally driven by the bleeding edge.

Reply:
Right, the behavior of older JITs may be different, but we do not really care. Note that we are paying a huge penalty by using the slow Span implementation on .NET Framework and this is peanuts compared to that.

Reply:
Ah, I see Jan beat me to it by 14 hours :)

Reply:
Addressed in b2f8026a93af711e63c60d1fa9e8c621ac5a61d2. Turns out there were lots of `ThrowHelper.Throw` methods marked `NoInlining`. I guess it might make sense to audit the wider libraries and make bulk changes.